src/ServiceControl/Hosting/Commands/CustomLivetimes/PersisterInitializingConsoleLifetime.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Hosting.Internal;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using ServiceControl.Persistence;
+
+sealed class PersisterInitializingConsoleLifetime : ConsoleLifetime, IHostLifetime
+{
+    readonly IPersistenceLifecycle persistenceLifecycle;
+
+    public PersisterInitializingConsoleLifetime(
+        IPersistenceLifecycle persistenceLifecycle,
```

Comment:
This is very subjective but I would have added all be base constructor parameters first and then the lifecycle one as an addition last.